### DISCUSSION TITLE
Tracking & Recording social provider when using Shield-oauth

### DISCUSSION THREAD

**Discussion by jlwegaba** [datamweb/shield-oauth#40] [Ideas]
Hi @datamweb and team,

Thanks for the awesome addition on Shield library.

I want to be able to track a social login provider say "Google", "Yahoo", or "Facebook". In the users table i have already a field column "social_provider".

How can I be able to add this on users data am saving in "users" table?

Thank you

**Comment by datamweb** [comment]
Hello, this is not possible at the moment.
The best way to do this is to add an `event` to **shield-oauth**.
I will try to add this possibility.

**Reply by jlwegaba** [reply]
Thank you @datamweb for quick response.

Please guide me on how i can add event in shield-oauth to be able to save the social_provider. 

Can I be able to split public array $usersColumnsName = [ ] as per the provider so that I can add it statically there?

**Comment by datamweb** [comment]
> Can I be able to split public array $usersColumnsName = [ ] as per the provider so that I can add it statically there?

At a glance, yes, you can do this.
Don't forget to add it(`social_provider`) in the section below:

https://github.com/datamweb/shield-oauth/blob/8b82c5bb9a03f6d7cec14d5429a6213abb23eff9/src/Config/ShieldOAuthConfig.php#L56-L60

https://github.com/datamweb/shield-oauth/blob/8b82c5bb9a03f6d7cec14d5429a6213abb23eff9/src/Models/ShieldOAuthModel.php#L24-L28

https://github.com/datamweb/shield-oauth/blob/8b82c5bb9a03f6d7cec14d5429a6213abb23eff9/src/Libraries/GoogleOAuth.php#L92-L115

**Reply by jlwegaba** [reply]
This worked perfectly. Thank you @datamweb 

Also edited the `$this->config->usersColumnsName['first_name'] => $userInfo->name,`  to `$this->config->usersColumnsName['first_name'] => $userInfo->given_name,` as that is how google calls the first_name under **shield-oauth/src/Libraries/GoogleOAuth.php**

Thank you

**Reply by evansharp** [reply]
column fix address permanently by PR #189

**Comment by datamweb** [comment]
> Please guide me on how i can add event in shield-oauth to be able to save the social_provider.

The event should be added in file [OAuthController](https://github.com/datamweb/shield-oauth/blob/develop/src/Controllers/OAuthController.php).

After that, you can easily perform any function after logging in with social media. For example, if it is entered with GoogleOAUTH, record it in the table and...
